src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -604,6 +611,48 @@ public function formAction($id, Request $request)
         ];
     }
 
+    /**
+     * Duplicate product info from one shop to another
+     *
+     * @param Product $product
+     * @throws \LogicException
+     */
+    private function duplicateProductFromDefaultShop(Product $product)
+    {
+        $productAdapter = $this->get('prestashop.adapter.data_provider.product');
+        $product = $productAdapter->getProduct($product->id, null, null, $product->id_shop_default);
+        $this->duplicateProductCombinationsFromDefaultShop($product);
+
+        return $product;
+    }
+    /**
+     * Duplicate combinations from one shop to another
+     *
+     * @param Product $product
+     */
+    private function duplicateProductCombinationsFromDefaultShop(Product $product)
+    {
+        $combinations = Product::getProductAttributesIds($product->id);
+        if ($combinations) {
+            foreach ($combinations as $id_combination) {
+                $combination = new Combination((int)$id_combination['id_product_attribute']);
+
+                $default_combination = new Combination(
+                    (int)$id_combination['id_product_attribute'],
+                    null,
+                    (int)$product->id_shop_default
+                );
+
+                $def = ObjectModel::getDefinition($default_combination);
```

Comment:
it's a really low level call: shouldn't we add a new method to Combination instead so we keep the complexity in the model?